x/mongo/driver/batch_cursor.go

```diff
@@ -155,12 +158,21 @@ func NewCursorResponse(info ResponseInfo) (CursorResponse, error) {
 type CursorOptions struct {
 	BatchSize             int32
 	Comment               bsoncore.Value
-	MaxTimeMS             int64
 	Limit                 int32
 	CommandMonitor        *event.CommandMonitor
 	Crypt                 Crypt
 	ServerAPI             *ServerAPIOptions
 	MarshalValueEncoderFn func(io.Writer) (*bson.Encoder, error)
+
+	// MaxAwaitTime is only valid for tailable awaitData cursors. If this option
+	// is set, it will be used as the "maxTimeMS" field on getMore commands.
+	MaxAwaitTime *time.Duration
+}
+
+// SetMaxAwaitTime will set the maxTimeMS value on getMore commands for
+// tailable awaitData cursors.
+func (cursorOptions *CursorOptions) SetMaxAwaitTime(dur time.Duration) {
```

Comment:
The cursor must still support maxAwaitTime per the CSOT specifications: https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.md#tailable-awaitdata-cursors